File: server/src/main/java/com/example/schoolsite/controllers/AutorizationController.java
package com.example.schoolsite.controllers;

import com.example.schoolsite.pojo.JwtResponse;
import com.example.schoolsite.pojo.LoginRequest;
import com.example.schoolsite.pojo.MessageResponse;
import com.example.schoolsite.pojo.SignUpRequest;
import com.example.schoolsite.services.AuthorizationService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@CrossOrigin(origins = "http://localhost:4200")
@RequestMapping("/api/auth")
@RequiredArgsConstructor
public class AutorizationController {

    private final AuthorizationService authorizationService;

    @PostMapping("/signIn")
    public ResponseEntity<?> authUser(@RequestBody LoginRequest loginRequest) {
        JwtResponse response = authorizationService.authUser(loginRequest);
        if(response!=null) {
            return ResponseEntity.ok(response);
        } else {
            return ResponseEntity
                    .badRequest()
                    .body(new MessageResponse("Error: Your login or password are invalid"));
        }
    }

    @PostMapping("/signUp")
    public ResponseEntity<?> registerUser(@RequestBody SignUpRequest signupRequest) {
        String status = authorizationService.registerUser(signupRequest);
        if(status.contains("Error")){
            return ResponseEntity
                    .badRequest()
                    .body(new MessageResponse(status));
        } else {
            return ResponseEntity.ok(new MessageResponse(status));
        }
    }
}

File: server/src/main/java/com/example/schoolsite/controllers/AutorizationController.kt
package com.example.schoolsite.controllers

import com.example.schoolsite.pojo.JwtResponse

@RestController
@CrossOrigin(origins = "http://localhost:4200")
@RequestMapping("/api/auth")
@RequiredArgsConstructor
class AutorizationController {
    private val authorizationService: AuthorizationService? = null
    @PostMapping("/signIn")
    fun authUser(@RequestBody loginRequest: LoginRequest?): ResponseEntity<*> {
        val response: JwtResponse = authorizationService.authUser(loginRequest)
        return if (response != null) {
            ResponseEntity.ok(MessageResponse())
        } else {
            ResponseEntity
                    .badRequest()
                    .body(MessageResponse("Error: Your login or password are invalid"))
        }
    }

    @PostMapping("/signUp")
    fun registerUser(@RequestBody signupRequest: SignUpRequest?): ResponseEntity<*> {
        val status: String = authorizationService.registerUser(signupRequest)
        return if (status.contains("Error")) {
            ResponseEntity
                    .badRequest()
                    .body(MessageResponse(status))
        } else {
            ResponseEntity.ok(MessageResponse(status))
        }
    }
}